packages/flutter/lib/src/foundation/basic_types.dart

```diff
@@ -112,19 +112,6 @@ class CachingIterable<E> extends IterableBase<E> {
   /// guaranteed that the underlying data set will only be walked
   /// once. If you have an [Iterable], you can pass its [iterator]
   /// field as the argument to this constructor.
-  ///
-  /// You can use a `sync*` function with this as follows:
-  ///
-  /// ```dart
-  /// Iterable<int> range(int start, int end) sync* {
-  ///   for (int index = start; index <= end; index += 1)
-  ///     yield index;
-  ///  }
-  ///
-  /// Iterable<int> i = CachingIterable<int>(range(1, 5).iterator);
-  /// print(i.length); // walks the list
-  /// print(i.length); // efficient
-  /// ```
```

Comment:
maybe explictly say to avoid using sync* with this, and explain why?

Reply:
Actually I should probably add this back. I was mis-reading it as encouraging people to write `sync*` methods, rather than how to use it with them. I'll add it back with some advice to avoid writing sync* though.